nilearn/datasets.py

```diff
@@ -26,6 +25,17 @@
 
 import nibabel
 
+# Define _get_time, which is used module-wide to get the current time.
+try:
+    # Use uptime (when available) to avoid reporting time while computer slept.
+    import uptime
+    if uptime.uptime() is None:
+        raise Exception("uptime is not supported by this OS.")
+    _get_time = uptime.uptime
```

Comment:
> it's not like the dataset download time is a critical piece of information and I must say I have probably never paid attention to it ...

I think that's the point: you never needed it so you never paid attention. But when you want to know (typically when you have a slow connection), having false information is very annoying. And this is not an edge case: when downloading a big dataset, I put my computer to sleep a lot (I do it 3 times a day: when commuting and when sleeping). Also, when I wanted to get my bandwith back temporary, I paused the process using Ctrl + Z, this also invalidate estimated time. Because of that, I ended up using wget instead of nilearn.
But I agree: this solution is not optimal (I thought that uptime was a standard module as it was installed on my box) and this is not the most important bug. Do we have a solution that can be coded fast and solve this problem ?